plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx

```diff
@@ -170,38 +135,80 @@ export const EntityOwnerPicker = () => {
           multiple
           disableCloseOnSelect
           loading={loading}
-          options={ownerEntities || []}
-          value={
-            ownerEntities?.filter(e =>
```

Comment:
Is it possible to keep the existing code around rendering the selected options (and using humanizeEntity)? Looks like this is just using the entity refs. Would make sense that it's not possible if we don't have the full list of read entities at load time.

Reply:
the entities are always humanized as before. The only case where aren't humanized is when they are selected (displayed as chips), to make it more explicit about which entity is selected (in case multiple ones have the same name)

Reply:
It might just be that I'm on a small screen but the chips just show the kind and namespace in the default view. The name is cutoff.

<img width="214" alt="Screenshot 2023-04-26 at 9 36 41 AM" src="https://user-images.githubusercontent.com/34432188/234592841-2e7ad677-fed1-49cb-979f-f32314ffd976.png">

Can we add an override of the `renderTags` option as well? Something like
```
renderTags={(tagValue, getTagProps) => {
            return tagValue.map((option, index) => {
              const isGroup =
                (typeof option === 'string'
                  ? parseEntityRef(option).kind
                  : option.kind
                ).toLocaleLowerCase('en-US') === 'group';
              return (
                <Chip
                  {...getTagProps({ index })}
                  icon={
                    isGroup ? (
                      <GroupIcon fontSize="small" />
                    ) : (
                      <PersonIcon fontSize="small" />
                    )
                  }
                  variant="outlined"
                  label={option?.metadata?.name || parseEntityRef(option).name}
                />
              );
            });
          }}
```

Reply:
it feels a bit weird to be to display the humanize entity instead of the entityRef here, since the intent is to filter by owner. Same in the dropdown. If it was me I would keep entityRef for everything, otherwise it might be problematic in case multiple entities have the same display name, or for adopters using namespaces

Reply:
It seems odd to me to have a different user experience for both though. 
The primary thing I was trying to solve with the original PR was that my org's team identifiers were all basically GUIDs and you couldn't easily search or filter them without knowing the exact value. Having the chips use the entity refs leaves the problem of how do I remove an item I selected if I know the display name but not the entity name.

Reply:
I'm feeling it would be best if the selected items look the same as the items that you select tbh, essentially keeping it as it is. It would be sweet to add the user/group icon to the chip though I think.

Reply:
Also, I've opened #17600 if we want to group by namespace in the results as part of this work or an additional PR afterwards.

Reply:
Fixed in https://github.com/backstage/backstage/pull/17514/commits/999fb3543325a0be6dee48c3ebe31b193bea65bb